libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -408,6 +409,17 @@ func setupInternalNetworkRules(bridgeIface string, addr *net.IPNet, icc, insert
 	var version iptables.IPVersion
 	var inDropRule, outDropRule iptRule
 
+	// Either add or remove the interface from the firewalld zone, if firewalld is running.
```

Comment:
It seems `ProgramChain` is only called by the bridge driver. Do you think it'd make sense to move the calls to Add/Del methods out of the `iptables` package to `setupIPTables` (ie. `setupInternalNetworkRules` caller), such that we don't duplicate these in two different places?

Reply:
Yes, I don't think the firewalld calls belong with the rest of `ProgramChain`, and should be called separately (and once) by the bridge driver.

It looked like a bit of a mess though. And, particularly for firewalld, not well regression tested. So, I figured a minimal change would be more likely to make it into 25.0.3.

And, it's the usual problem with changing an exported package? Maybe that doesn't matter? Or, if it does, we could split `ProgramChain`, and make it call new functions that we use from the bridge driver?

But perhaps that should be a separate refactoring PR?

Reply:
Excellent! Thanks Cory.